Battleship/app/src/main/java/by/belotskiy/battleship/activity/GameActivity.java

```diff
@@ -0,0 +1,339 @@
+package by.belotskiy.battleship.activity;
+
+import android.content.pm.ActivityInfo;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+import com.google.firebase.database.ChildEventListener;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+import com.google.firebase.storage.FirebaseStorage;
+import com.squareup.picasso.Picasso;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Locale;
+
+import by.belotskiy.battleship.R;
+import by.belotskiy.battleship.game.Cell;
+import by.belotskiy.battleship.game.Field;
+import by.belotskiy.battleship.game.Fleet;
+import by.belotskiy.battleship.game.Ship;
+import by.belotskiy.battleship.game.enums.CellType;
+import by.belotskiy.battleship.game.enums.FieldType;
+import by.belotskiy.battleship.game.service.FleetService;
+
+public class GameActivity extends AppCompatActivity {
+    private final String WIN = "Win";
+    private final String LOSE = "Lose";
+    private final String CURRENT_PLAYER = "CurrentPlayer";
+    private FirebaseDatabase database = FirebaseDatabase.getInstance();
+    private DatabaseReference roomRef;
+    private DatabaseReference profileRef;
+    private DatabaseReference statisticsRef;
```

Comment:
Now I will ask you to extract DB interaction and game logic from the Activity (which is UI layer).